openqa-label-known-issues

```diff
@@ -18,17 +18,21 @@ trap 'test "$KEEP_REPORT_FILE" == "1" || rm "$out"' EXIT
 echoerr() { echo "$@" >&2; }
 
 label_on_issue() {
-    id=$1
-    search_term=$2
-    comment=$3
-    restart=$4
-    grep_opts="${grep_opts:-"-qPzo"}"
-    check=$5
+    local id=$1 search_term=$2 comment=$3 restart=$4 check=$5
+    local rc=0 grep_output
+    local grep_opts="${grep_opts:-"-qPzo"}"
     # shellcheck disable=SC2086
     if [[ -n $check ]]; then
         eval $check || return 1
     else
-        grep $grep_opts "$search_term" "$out" || return 1
+        grep_output=$(grep $grep_opts "$search_term" "$out" 2>&1) || rc=$?
+        if (( "$rc" == 1 )); then
+            return 1
+        elif (( "$rc" != 0 )); then
+            # unexpected error, e.g. "exceeded PCRE's backtracking limit"
```

Comment:
I have seen this specific error as well but I am not sure if we need to do something about it. I have seen cases where despite this notice we have successfully labeled openQA jobs. Please try `echo https://openqa.opensuse.org/tests/1859482 | dry_run=1 ./openqa-label-known-issues` before and after your change

Reply:
before:
```
grep: exceeded PCRE's backtracking limit                   
openqa-cli api --host https://openqa.opensuse.org -X POST jobs/1859482/comments text=poo#80628 [qe-core][opensuse][upgrade][sporadic][needs-refining] firefox: very frequently does not show the top menu bar auto_review:"(?s)HDD_1.*opensus
e-42.*no candidate needle.*firefox-top-bar-highlighted.*matched"
```
after:
```
grep failed: cmd=>grep -qPzo '(?s)2021-.*T.*Error connecting to <root@s390p.*.suse.de>: No route to host' '/tmp/tmp.FL3ntlzAdE'< output='grep: exceeded PCRE's backtracking limit'
openqa-cli api --host https://openqa.opensuse.org -X POST jobs/1859482/comments text=poo#80628 [qe-core][opensuse][upgrade][sporadic][needs-refining] firefox: very frequently does not show the top menu bar auto_review:"(?s)HDD_1.*opensus
e-42.*no candidate needle.*firefox-top-bar-highlighted.*matched"
```
> I am not sure if we need to do something about it

There is an error message in the log from grep, and anyone looking at the log does not know where it comes from. Well, we know now, but others might not.
How can people decide if this is something that should be handled?
What's bad about making the error message more informative?

The original task was "No obvious error messages in gru logs".

Reply:
Oh, and thanks for indirectly letting me know how the script is supposed to be called, because the variable name `i` did not make it very guessable what the input is.

Reply:
this looks good, thanks. Of course it's good if you improve the error messages. I just wanted to clarify if that has an impact on how we label jobs.